spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/batch/JobService.java

```diff
@@ -349,4 +349,16 @@ StepExecution getStepExecution(Long jobExecutionId, Long stepExecutionId) throws
 	 */
 	Collection<JobExecution> listJobExecutionsForJob(String jobName, BatchStatus status, int pageOffset, int pageSize)
 			throws NoSuchJobException;
+
+	/**
+	 * List the {@link JobExecutionWithStepCount JobExecutions} in descending order of
+	 * creation (usually close to execution order) without step execution data filtered by job
+	 * names matching query string.
+	 *
+	 * @param queryString search query string to filter job names
+	 * @param start the index of the first execution to return
+	 * @param count the maximum number of executions
+	 * @return a collection of {@link JobExecutionWithStepCount}
+	 */
+	Collection<JobExecutionWithStepCount> listFilteredJobExecutionsWithStepCount(String queryString, int start, int count);
```

Comment:
I concerned that there maybe a case for sql injection here.

Reply:
If I'm not wrong, we are using JDBCTemplate here which internally uses PreparedStatements which protects it from SQL Injections. So I think this should be safe.

Reply:
Changed that to jobName